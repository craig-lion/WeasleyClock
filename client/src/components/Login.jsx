import React, {useState, useEffect} from 'react';
import axios from 'axios';
import styled from 'styled-components';

const Login = (props) => {

  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');


  const handleSubmit = (e) => {
    e.preventDefault();
    let query = {userName, password};
    axios.post('/api/login', query)
    .then((result) => {
      if(result.data === false) {alert('Login Failed')}
      else {
      console.log('this is results.data in Login: ', result.data)
      let sendUser = async () => {
        props.setLoginUserName(result.data)
      }
      sendUser().then(props.setIsLoggedIn(true));
      }
    })
    setUserName('');
    setPassword('');
  }

  const handleNewUser = (e) => {
    e.preventDefault();
    let query = {userName, password};
    axios.post('/api/addUser', query)
    .then((result) => {
      // console.log('this is result: ', result)
      props.setIsLoggedIn(true);
    })
    setUserName('');
    setPassword('');
  }

  return (
    <LoginStyle>
    <Title> New Wiz WhoDis?</Title>
    <form>
    <Div>
      <Label>What is your Wizard Name?</Label>
    </Div>
  <Text type="text" color="red" id="userName" value={userName} onChange={e => setUserName(e.target.value)} name="userName"></Text>
  <Div>
      <Label>What is your Secret Spell?</Label>
    </Div>
  <Text type="text" color='antiquewhite' id="password" value={password} onChange={e => setPassword(e.target.value)} name="password"></Text>
  <Div>
  <Button type="submit" value="Y'all Know Who I Am" onClick={handleSubmit}></Button>
  </Div>
  <Div>
  <Button type="submit" value="New Wiz" onClick={handleNewUser}></Button>
  </Div>
  </form>
  </LoginStyle>
  )
}

const Title = styled.p`
  margin:0px;
  font-size:100px;
`

const Div = styled.div`
  padding:5px
  text-align: center;
  text-align-last: center;
  font-size:30px;
  -moz-text-align-last: center;
  font-family: 'Luminari';
`

const LoginStyle = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vw;
  background-image: url('darkWood.jpg');
  text-shadow: -2px -2px 0 #000, 2px -2px 0 #000, -2px 2px 0 #000, 2px 2px 0 #000;
  `

const Text = styled.input`
  background-image: url('darkWood.jpg');
  border-radius:18px;
  border-color:antiqueWhite;
  text-align: center;
  text-align-last: center;
  -moz-text-align-last: center;
  font-family: 'Luminari';
`

const Label = styled.label`
  color:antiquewhite;
  font-size:12px;
`

const Button = styled.input`  
  color:AntiqueWhite;
  background-image: url('darkWood.jpg');
  opacity: 50%;
  border-radius:18px;
`

export default Login;