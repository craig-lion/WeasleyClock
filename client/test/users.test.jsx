import 'regenerator-runtime/runtime';
import {
  describe, expect, it, beforeEach, afterEach, jest,
} from '@jest/globals';
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import axios from 'axios';
import Login from '../src/components/Login';

const domTestingLib = require('@testing-library/dom');

const { queryHelpers } = domTestingLib;

jest.mock('axios');

describe('users', () => {
  let wrapper;

  describe('authentication', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);
    const postPromise = Promise.resolve({ data: true });
    axios.post.mockImplementation(() => postPromise);
    const setLoginUserName = jest.fn();
    const setIsLoggedIn = jest.fn();
    // const { asFragment } = render(<Login
    //   setLoginUserName={setLoginUserName}
    //   setIsLoggedIn={setIsLoggedIn}
    // />);

    // it('FragmentTest', () => {

    //   expect(asFragment()).toMatchSnapshot();
    // });

    beforeEach(() => {
      // wrapper = Enzyme.mount((<Login
      //   setLoginUserName={setLoginUserName}
      //   setIsLoggedIn={setIsLoggedIn}
      // />));
      wrapper = render(<Login
        setLoginUserName={setLoginUserName}
        setIsLoggedIn={setIsLoggedIn}
      />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('can create new user', async () => {
      const getById = queryHelpers.queryByAttribute.bind(null, 'id');
      fireEvent.click(getById(wrapper.container, 'newUser'));
      expect(axios.post).toHaveBeenCalledWith('/api/addUser', { userName: expect.any(String), password: expect.any(String) });
      await postPromise;
      expect(setLoginUserName).toHaveBeenCalledWith(expect.any(String));
      expect(setIsLoggedIn).toHaveBeenCalledWith(true);
    });

    // it('existing user can login', async () => {
    //   const knownUser = wrapper.find('#knownUser').at(0);
    //   knownUser.simulate('click');
    //   expect(axios.post).toHaveBeenCalledWith('/api/login', { userName: expect.any(String), password: expect.any(String) });
    //   await postPromise;
    //   expect(setLoginUserName).toHaveBeenCalledWith(expect.any(String));
    //   expect(setIsLoggedIn).toHaveBeenCalledWith(true);

    //   const userName = wrapper.find('#userName').at(0);
    //   const password = wrapper.find('#password').at(0);

    //   userName.value = 'TestUser';
    //   userName.simulate('change', { target: userName });
    //   expect(setState).toHaveBeenCalledWith('TestUser');

    //   password.value = 'TestPass';
    //   password.simulate('change', { target: password });
    //   expect(setState).toHaveBeenCalledWith('TestPass');
    // });
  });
  // describe('friends', () => {

  // });
});

// describe("Title input", () => {
//   it("Should capture title correctly onChange", () => {
//       const title = wrapper.find("input").at(0);
//       title.instance().value = "Test";
//       title.simulate("change");
//       expect(setState).toHaveBeenCalledWith("Test");
//   });
// });
