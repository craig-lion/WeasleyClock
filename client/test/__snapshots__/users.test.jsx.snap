// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw iyuiXf"
  >
    <p
      class="sc-AxjAm krrDgX"
    >
       New Wiz WhoDis?
    </p>
    <form>
      <div
        class="sc-AxirZ cfvkjE"
      >
        <label
          class="sc-AxhUy debCfc"
        >
          What is your Wizard Name?
        </label>
      </div>
      <input
        class="sc-AxhCb eYTsEC"
        color="red"
        id="userName"
        name="userName"
        type="text"
        value=""
      />
      <div
        class="sc-AxirZ cfvkjE"
      >
        <label
          class="sc-AxhUy debCfc"
        >
          What is your Secret Spell?
        </label>
      </div>
      <input
        class="sc-AxhCb eYTsEC"
        color="antiquewhite"
        id="password"
        name="password"
        type="text"
        value=""
      />
      <div
        class="sc-AxirZ cfvkjE"
      >
        <input
          class="sc-AxgMl eqWaMp"
          id="knownUser"
          type="submit"
          value="Known Wizard"
        />
      </div>
      <div
        class="sc-AxirZ cfvkjE"
      >
        <input
          class="sc-AxgMl eqWaMp"
          id="newUser"
          type="submit"
          value="New Wizard"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`users authentication FragmentTest 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw iyuiXf"
  >
    <p
      class="sc-AxjAm krrDgX"
    >
       New Wiz WhoDis?
    </p>
    <form>
      <div
        class="sc-AxirZ cfvkjE"
      >
        <label
          class="sc-AxhUy debCfc"
        >
          What is your Wizard Name?
        </label>
      </div>
      <input
        class="sc-AxhCb eYTsEC"
        color="red"
        id="userName"
        name="userName"
        type="text"
        value=""
      />
      <div
        class="sc-AxirZ cfvkjE"
      >
        <label
          class="sc-AxhUy debCfc"
        >
          What is your Secret Spell?
        </label>
      </div>
      <input
        class="sc-AxhCb eYTsEC"
        color="antiquewhite"
        id="password"
        name="password"
        type="text"
        value=""
      />
      <div
        class="sc-AxirZ cfvkjE"
      >
        <input
          class="sc-AxgMl eqWaMp"
          id="knownUser"
          type="submit"
          value="Known Wizard"
        />
      </div>
      <div
        class="sc-AxirZ cfvkjE"
      >
        <input
          class="sc-AxgMl eqWaMp"
          id="newUser"
          type="submit"
          value="New Wizard"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
